lib/storage/src/rbac/mod.rs

```diff
@@ -210,44 +203,61 @@ impl std::fmt::Display for CollectionPass<'_> {
     }
 }
 
-#[derive(Eq, PartialEq, Copy, Clone, Debug)]
-enum AccessMode {
-    Read,
-    ReadWrite,
-    Manage,
+#[derive(Default, Debug, Copy, Clone)]
+pub struct AccessRequrements {
+    /// Write access is required.
+    pub write: bool,
+    /// Manage access is required, implies write access.
+    pub manage: bool,
```

Comment:
Why not use the `Access` enum itself?

Reply:
It's introduced to address this comment: https://github.com/qdrant/qdrant/pull/3941#discussion_r1549325924
Precisely, `Requirements` is added to avoid comparing two enums.